libs/state/docs/snippets/model-and-viewModel-with-rx-tate.md

```diff
@@ -0,0 +1,259 @@
+In this example we will see what is the deifference between Model and ViewModel and show different approaches how Model & ViewModel can be handled using RxState. First we will go with imperative approach and after that make it fully reactive.
+
+# What is what?
+### What is Model?
+Model is an object or class that holds our data.
+### What is ViewModel?
+ViewModel is data that is needed in our views.
+### What is View?
+View is our component template.
+
+# Intial app setup
+In this example we will take a look at a simple list app. We will focus on next things:
+- `List` component that renders list name, items count and items. List can be open or closed.
+- `ListService` is a service that will communicate with our API layer and get list data.
+
+For this example we will skip any additional logic such as routing, error handling etc.
+
+### Interfaces
+
+We have 2 interfaces 
+- `IList` that represents the model that we recieving from API 
+- `IViewList` represents a model that we actually need in our component template.
+
+```
+interface IList {
+  id: string;
+  name: string;
+  items: string[];
+}
+```
+```
+interface IViewList {
+  name: string;
+  items: string[];
+  itemsTotal: number;
+  isOpen: boolean;
+}
+```
+As you can see they are a bit different. Not all `IList` data needed in the View and at the same time `IViewList` contains some additional properties that are not in the `IList`.
+
+### ListService
```

Comment:
I would focus on the concept of model vs view model. A service here can maybe named by I see the code as overhead.
In general, we should shrink it down to its essence that is related to the model vs view model.

Reply:
Will remove service code and try to reduce this wall of text a bit. For now will convert this request to draft.

Reply:
But don't throw it away, it's good stuff there.